include/irr/asset/IShader.h

```diff
@@ -32,7 +32,7 @@ class IShader
 			auto findLineJustAfterVersionOrPragmaShaderStageDirective = [&_glsl]
 			{
 				size_t hashPos = _glsl.find_first_of('#');
-				if (_glsl.compare(0, 8, "#version", hashPos, 8))
+				if (_glsl.compare(hashPos, 8, "#version"))
```

Comment:
make sure it won't throw (that you still have 8 characters in the string left).

Reply:
it checks sizes by itself

Reply:
yes, and then it throws according to C++ spec?

Reply:
no, only in case of offset parameters, lengths are taken care of. https://en.cppreference.com/w/cpp/string/basic_string/compare variant (5)

>Compares a [pos1, pos1+count1) substring of this string to the null-terminated character sequence beginning at the character pointed to by s with length Traits::length(s) **If count1 > size() - pos1 the substring is [pos1, size()).**

Reply:
Funny cause that same page says **Exceptions** and notes that it will throw is `pos1` is out of range.

So check and avoid a throw.

Reply:
`pos1` is first param so it makes sense to check in case we assume there might be `_glsl` string with no '#' in it. If thats what u meant i just got confused by "(that you still have 8 characters in the string left)"

Reply:
bascailly make sure that it does not throw no matter what `_glsl` string we pass to it, empty or shorter than 8 characters, etc.